### ISSUE TITLE
Doesn't work with Raspberry Pi 4b

### ISSUE THREAD

**Issue by zDxrkXeno** [sphero-inc/sphero-sdk-raspberrypi-python#20]
I have installed the Sphero SDK on my Raspberry Pi 4, when I run the program it says "Checking RVR firmware versions..." then quits without running it. What should I do?

**Comment by zDxrkXeno** [comment]
@Jim-Konish Any ideas?

**Comment by Jim-Konish** [comment]
We tested on the Pi 4 and didn't have any issues after making the updates to the setup process that we released in December.  

If you have not connected RVR to the Sphero EDU app, you will need to do so to update the firmware.

Most connection problems are a result of bad wiring, either with jumpers connected incorrectly, or with damage (often not visible) to the jumpers themselves so that they do not electrically connect.  Recheck your wiring, [following these instructions](https://sdk.sphero.com/docs/getting_started/raspberry_pi/raspberry_pi_hardware_setup), and if it still doesn't work try different jumper wires.

The next troubleshooting step is to perform a loopback test to verify the Pi UART is working properly, as described [here](https://amitasinghchauhan.medium.com/serial-port-debugging-101-loopback-test-4a7e40da9055).  I have seen UARTs go bad on a Pi before.

Finally, if you are still having trouble and you cloned the SDK from GitHub, it is worth following the [quick start instructions](https://sdk.sphero.com/docs/getting_started/raspberry_pi/raspberry_pi_setup_preconfigured) to see if there is some other configuration issue in your Pi image.  The quick start uses a pre-configured SD card image that requires no extra setup.

**Comment by zDxrkXeno** [comment]
I tried running the minicom but when I typed I got no response. Is my Pi bad? I have only had it for a couple of months. @Jim-Konish

**Comment by Jim-Konish** [comment]
It's possible the Pi has a bad UART, or it might be configured incorrectly.  You may be able to look for other resources on Pi UART troubleshooting to figure that out.

Did you try the quick start instructions?  That image should have the UART configured correctly, so it's worth a shot.

**Comment by zDxrkXeno** [comment]
> It's possible the Pi has a bad UART, or it might be configured incorrectly.  You may be able to look for other resources on Pi UART troubleshooting to figure that out.
> 
> 
> 
> Did you try the quick start instructions?  That image should have the UART configured correctly, so it's worth a shot.

Yes I followed the instructions and tried the image no luck. I would have to find another way to test the UART.
